plugin/core/sessions.py

```diff
@@ -943,12 +943,16 @@ def on_session_buffer_changed_async(self, session_buffer: SessionBufferProtocol)
         """
         pass
 
-    def on_session_end_async(self) -> None:
+    def on_session_end_async(self, exit_code: Optional[int], exception: Optional[Exception]) -> None:
         """
         Notifies about the session ending (also if the session has crashed). Provides an opportunity to clean up
         any stored state or delete references to the session or plugin instance that would otherwise prevent the
-        instance from being garbage-collected. If the plugin hasn't crashed, a shutdown message will be send immediately
-        after this method returns.
+        instance from being garbage-collected.
+
+        If the session hasn't crashed, a shutdown message will be send immediately
+        after this method returns. In this case exit_code and exception are None.
+        If the session has crashed, the exit_code or the exception is not None.
```

Comment:
I suppose it could be more precise. In case of a crash `exit_code` will always be set and `exception` is optional.

```suggestion
        If the session has crashed, the exit_code and an optional exception are provided.
```